### PR TITLE
updating shutdown hook to write failures to a log file and updating the test to read from the log file if one exists

### PR DESCRIPTION
**Pull request by jacobwdv**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

**Comment by jacobwdv** [comment]
#build

**Comment by jacobwdv** [comment]
@OpenLiberty/admin -- this change would create a logfile named shutdownHook.log if / when the shutdown hook fails. No log is written unless a failure occurs. This log is written so we can diagnose why a cleanup may have failed either in a test or user environment. 

I think this is a behaviour change in that a new log may appear, but it also doesn't change anything about the old logs / how the feature works.

**Comment by jacobwdv** [comment]
@build

**Comment by jacobwdv** [comment]
#build

**Comment by jacobwdv** [comment]
@donbourne -- I got in a conversation with Wendy and she said you might be the best person to ask about this PR. Basically, the behaviour I want to add will create a file named shutdownHook.log in the log directory for a packaged server ONLY when something goes wrong with the hook. This is part of an effort to get this test to pass as it has been failing intermitently in the pipeline for some time now. 

The change to the code also seems to fix the issue, we wait for the scripts to complete running before quitting and that seems to make a difference.

File: dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java [hunk 1]
```diff
@@ -286,22 +359,32 @@ private File writeCleanupFile(int platformType) throws IOException {
...
 
     }
 
+    /**
+     * @return
+     */
+    private Path getHookLog() {
+        String logDir = dir + File.separator + "wlp" + File.separator + "usr" + File.separator + "servers" + File.separator + serverName + File.separator + "logs";
+        return Paths.get(logDir, hookLog);
+    }
```
Comment (donbourne) [inline dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java:388]: this doesn't look like it accounts for possibility that server is running with an explicitly set LOG_DIR or non-default server directory location.

Comment (jacobwdv) [reply dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java:388]: I don't believe this is settable with wikth the packaged server code. Spoke with Don over slack about it. 

@hlhoots / @bensonlatibm can you confirm? I don't think you can change the log directly when we package the server up in a jar file. At least the code never afforded for that i don't think.

**Review by hlhoots — DISMISSED** [review]
LGTM pending a good build.

**Review by bensonlatibm — DISMISSED** [review]
LGTM!

**Comment by jacobwdv** [comment]
#build

**Comment by jacobwdv** [comment]
#libby

**Comment by jacobwdv** [comment]
failures above are not from my changes. 
image

**Comment by jacobwdv** [comment]
#build

**Review by bensonlatibm — APPROVED** [review]
LGTM!

**Comment by jacobwdv** [comment]
failures unrelated to my changes

#libby

**Comment by jacobwdv** [comment]
#libby

**Comment by jacobwdv** [comment]
#link 295756

**Comment by jacobwdv** [comment]
RTC DEFECT: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.workitem.viewWorkItem&id=295756

**Pull request merged**